src/cli/main.cpp

```diff
@@ -52,6 +53,14 @@
 #define TOSTRING(x) STRINGIFY(x)
 #define VERSION_STR TOSTRING(VERSION)
 
+class TerminalReporter : public DiagnosticReporter {
+  public:
+    void report(Diagnostic diagnostic) {
+      std::cerr << diagnostic.getMessage() << std::endl;
+    }
+};
+DiagnosticReporter *reporter = new TerminalReporter();
```

Comment:
Does this need to be a pointer? I'd much rather just see:

```
TerminalReporter reporter;
```

Or does that not work?

Reply:
Have to resort to using a pointer because DiagnosticReporter is an abstract class